shelley/chain-and-ledger/shelley-spec-ledger-test/test/Test/Shelley/Spec/Ledger/Serialisation/Golden/Genesis.hs

```diff
@@ -39,19 +42,30 @@ import Test.Shelley.Spec.Ledger.Utils
     unsafeMkUnitInterval,
   )
 import Test.Tasty
+import Test.Tasty.HUnit (Assertion, testCase, (@?=))
 import Test.Tasty.Hedgehog
 
-prop_golden_ShelleyGenesis :: Property
-prop_golden_ShelleyGenesis = goldenTestJSONPretty example "test/Golden/ShelleyGenesis"
+prop_golden_json_ShelleyGenesis :: Property
+prop_golden_json_ShelleyGenesis = goldenTestJSONPretty example "test/Golden/ShelleyGenesis"
   where
     example :: ShelleyGenesis C
     example = exampleShelleyGenesis
 
+prop_golden_cbor_ShelleyGenesis :: Assertion
+prop_golden_cbor_ShelleyGenesis =
+  Base16.encode (serialize example) @?= b16
+  where
+    example :: ShelleyGenesis C
+    example = exampleShelleyGenesis
+
+    b16 = "83a50081824a9db8a41713ad20245f4e00018182510075c40f44e1c155bedab80d3ec7c2190b0a02185e030a05a149e0dab80d3ec7c2190b1864a10082824873ed39075e40d2a6507b208423b1e15f5173ed39075e40d2a682489ac25c873e2248cc507b208423b1e15f519ac25c873e2248ccf6"
```

Comment:
This works, but good luck finding out what changed when it fails :grin:

Reply:
yea, that's true. Perhaps we should compare the cbor tokens instead.

Reply:
We do something fancy in consensus: we store the bytes in a file, but convert to CBOR in case of a mismatch. We also deal with nested CBOR and actually return a diff of the CBOR. See https://github.com/input-output-hk/ouroboros-network/blob/95c9cd29ed889abe170ab2e9c6e8523b02432f03/ouroboros-consensus-test/src/Test/Util/Serialisation/Golden.hs#L87-L93

I don't think all that is needed here, though. Moreover, we really don't expect the `ShelleyGenesis` to change anymore :slightly_smiling_face:.

Reply:
yep, that's much nicer! In order to unblock https://github.com/input-output-hk/ouroboros-network/pull/2694 I think it makes sense to merge this once CI is happy. and we can make the tests failures easier to read if it becomes a problem. Indeed I do not expect the shelley genesis to change :)